test/CMakeLists.txt

```diff
@@ -0,0 +1,175 @@
+# All optional argment (i.e., the ones not being named) are automatically passed to the command being called.
+function(setup_test_serial_python TESTNAME TESTDIR TESTSCRIPT)
+  setup_test_dir(${TESTDIR})
+  add_test(
+    NAME ${TESTNAME}
+    COMMAND ${Python3_EXECUTABLE} ${TESTSCRIPT} --launch_cmd=${PROJECT_BINARY_DIR}/bin/enzo-e ${ARGN}
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/${TESTDIR})
+  set_tests_properties(${TESTNAME} PROPERTIES LABELS "serial;enzo;regression" )
+endfunction()
+
+# Parallel (using multiple compute elements) test wrapper functions calling a script (e.g., to include output analysis)
+# All optional argment (i.e., the ones not being named) are automatically passed to the command being called.
+function(setup_test_parallel_python TESTNAME TESTDIR TESTSCRIPT)
+  # construct a list of the various parts of the launch_cmd argument (this is
+  # necessary since PARALLEL_LAUNCHER_NPROC_ARG may itself be a list)
+  set(LAUNCH_CMD_LIST
+    ${PARALLEL_LAUNCHER}
+    ${PARALLEL_LAUNCHER_NPROC_ARG} ${PARALLEL_LAUNCHER_NPROC}
+    ${PROJECT_BINARY_DIR}/bin/enzo-e)
+  # now join the elements of LAUNCH_CMD_LIST into a single string so that the
+  # shell interprets the entire string as a single command line argument
+  LIST(JOIN LAUNCH_CMD_LIST " " LAUNCH_CMD)
+
+  setup_test_dir(${TESTDIR})
+  add_test(
+    NAME ${TESTNAME}
+    COMMAND ${Python3_EXECUTABLE} ${TESTSCRIPT} --launch_cmd ${LAUNCH_CMD} ${ARGN}
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/${TESTDIR})
+  set_tests_properties(${TESTNAME} PROPERTIES LABELS "parallel;enzo;regression" )
+endfunction()
+
+############################### UNIT TESTS ####################################
+# The following tests are self-contained binary of individual components
+
+setup_test_unit(Array ArrayComponent/Array test_cello_array)
+# TODO(need help) Need to reintroduce following tests once dependency hell is resolved,
+# see commented unit tests in src/Cello/CMakeLists.txt
+#setup_test_unit(CelloType Cello/Type test_type)
+#setup_test_unit(Data-Field DataComponent/Field test_field)
+#setup_test_unit(Data-Field-Data DataComponent/FieldData test_field_data)
+#setup_test_unit(Data-Field-Descr DataComponent/FieldDescr test_field_descr)
+#setup_test_unit(Data-Field-Face DataComponent/FieldFace test_field_face)
+#setup_test_unit(Data-Grouping DataComponent/Grouping test_grouping)
+#setup_test_unit(Data-ItIndex DataComponent/ItIndex test_itindex)
+#setup_test_unit(Data-Particle DataComponentParticle/ test_particle)
+#setup_test_unit(Data-Scalar DataComponent/Scalar test_scalar)
+#setup_test_unit(EnzoUnits UnitsComponent/EnzoUnits test_enzo_units)
+setup_test_unit(Error ErrorComponent/Error test_error)
+#setup_test_unit(Schedule IOComponent/Schedule test_schedule)
+#setup_test_unit(Colormap IOComponent/Colormap test_colormap)
+setup_test_unit(Memory MemoryComponent/Memory test_memory)
+setup_test_unit(Monitor MonitorComponent/Monitor test_monitor)
+#setup_test_unit( Component/ test_)
```

Comment:
It might be worthwhile to add something to the documentation about the currently missing unit tests.